internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -651,28 +686,66 @@ func (r *GitJobReconciler) newJobSpec(ctx context.Context, gitrepo *v1alpha1.Git
 	}
 
 	volumes, volumeMounts := volumes(configMap.Name)
+	var certVolCreated bool
 
 	if gitrepo.Spec.HelmSecretNameForPaths != "" {
-		vols, volMnts := volumesFromSecret(ctx, r.Client,
+		vols, volMnts, hasCertVol := volumesFromSecret(ctx, r.Client,
 			gitrepo.Namespace,
 			gitrepo.Spec.HelmSecretNameForPaths,
 			"helm-secret-by-path",
+			"",
 		)
 
+		certVolCreated = hasCertVol
+
 		volumes = append(volumes, vols...)
 		volumeMounts = append(volumeMounts, volMnts...)
 
 	} else if gitrepo.Spec.HelmSecretName != "" {
-		vols, volMnts := volumesFromSecret(ctx, r.Client,
+		vols, volMnts, hasCertVol := volumesFromSecret(ctx, r.Client,
 			gitrepo.Namespace,
 			gitrepo.Spec.HelmSecretName,
 			"helm-secret",
+			"",
 		)
 
+		certVolCreated = hasCertVol
+
 		volumes = append(volumes, vols...)
 		volumeMounts = append(volumeMounts, volMnts...)
 	}
 
+	// In case no Helm secret volume has been created, because Helm secrets don't exist or don't contain a CA
+	// bundle, mount a volume with a Rancher CA bundle.
+	if !certVolCreated {
+		// Fall back to Rancher-configured secrets
+		// We need to copy secret data from Rancher, because Rancher secrets live in a different namespace and
+		// can therefore not be used as sources for a volume.
+		secretName := fmt.Sprintf("%s-rancher-cabundle", gitrepo.Name)
```

Comment:
Nit: Maybe better to have these names in constants or a function? Similar to the `caBundleName` function.

Reply:
Moved this one to a separate function. I think that moving others to separate variables/functions could be done as part of a separate effort, refactoring building of git jobs.